docs/helpshift-widget/api.mdx

```diff
@@ -0,0 +1,433 @@
+
+Make sure you do not overwrite the `helpshiftWidgetConfig` object when trying to update it.
+
+</Admonition>
+
+<Admonition type="info" title="Note">
+
+1. This API is applicable to all the `helpshiftWidgetConfig` object options, such as user related data, language, full privacy, widgetOptions, widgetType, etc.
+2. Once the conversation has started, calling this API will not update the user identity or update Custom Issue Fields for an ongoing conversation.
+3. This API will cause the widget to reload. If the user has opened the widget and this API is called, then the user will see the loading indicator, and the conversation of the updated user will be displayed.
+
+</Admonition>
+
+## Set Initial User Message {#set-initial-user-message}
+
+If you want to set the initial user message through an API rather than making the end user type it, you can use the following API.
+
+```javascript
+var userMessage = "Hello, how can I track my order?";
+HelpshiftWidget("setInitialUserMessage", userMessage);
+```
+
+<Admonition type="info" title="Note">
+
+You should call this API before a conversation starts. In other words, it works only when the conversation is not open i.e. when a new user opens the widget or existing user has closed a conversation. If you call this API in the middle of an existing conversation it will not have any effect.
+
+</Admonition>
+
+## Events {#events}
+
+The Helpshift widget fires the following events across different scenarios. You can use the `addEventListener` and `removeEventListener` APIs to manage the events.
+
+### New Unread Messages Event {#new-unread-messages-event}
+
+This event gets fired when there are new unread message for the user. To listen to this event, add the following code after the embed code.
+
+<Admonition type="info" title="Note">
+
+This event can also be used to show a browser notification when the unread messages count changes.
+
+More info on [Browser notifications](https://developer.mozilla.org/en-US/docs/Web/API/notification)
+
+</Admonition>
+
+#### Adding the New Unread Messages event handler
+
+```javascript
+var newUnreadMessagesEventHandler = function (data) {
+  console.log("Unread count = ", data.unreadCount);
+};
+
+HelpshiftWidget(
+  "addEventListener",
+  "newUnreadMessages",
+  newUnreadMessagesEventHandler
+);
+```
+
+#### Removing the New Unread Messages event handler
+
+```javascript
+HelpshiftWidget(
+  "removeEventListener",
+  "newUnreadMessages",
+  newUnreadMessagesEventHandler
+);
+```
+
+### User Changed Event {#user-changed-event}
+
+This event gets fired when the user reopens Helpshift widget by clicking on the email link, if we detect that the user's identifier
+has changed from when the original user started the conversation to when this user clicked the email link.
+
+<Admonition type="info" title="Note">
+
+The ‘Follow up via email’ toggle for Helpshift widget needs to be enabled for you to use this event. The User Changed event can
```

Comment:
Re: The ‘Follow up via email’ toggle for Helpshift widget  -> The `Follow up via email` toggle is on the web chat config page on agent dashboard

Reply:
Done